lib/edition_tagger.rb

```diff
@@ -7,7 +7,10 @@ def initialize(edition_tag_associations, logger)
 
   def run
     @edition_tag_associations.each do |association|
-      add_mainstream_browse_tag(association[:slug], association[:tag])
+      unless ["TRUE", "FALSE", nil].include? association[:primary]
+        raise "Invalid value for `primary`: #{association[:primary]}"
```

Comment:
I expect a risk of invalid values because this is expected to be fed by a spreadsheet created manually, and humans make typos.  A test is a good idea, though: added https://github.com/alphagov/publisher/commit/faaf043aa167e302de9982ec441a0863ab367cf6#diff-af472a3dbebbe1ff56ff17d442aeedeaR85